lib/fs/dumbfs.c

```diff
@@ -42,32 +43,125 @@ dumbfs_unmount(void *cookie)
 
 static status_t
 dumbfs_open(
-	void *fscookie __unused /* TODO */,
-	char const *path __unused /* TODO */,
-	struct filehandle *handle __unused /* TODO */
+	struct file_handle *handle,
+	char const *path
 )
 {
-	/* TODO */
-	return (ERR_NOT_FOUND);
+	uint32 file_index;
+	struct dumbfs_file_entry file_header[1];
```

Comment:
I know you have your believes about 1-length arrays, but C is a language for real men, we are all grown-up, and therefore we shouldn't be afraid of using real pointer types even if it's not as safe as 1-length arrays :p

(To be honest, it's a pretty rare idiom that adds (and in this example in particular), pretty much nothing. I think it's better to stick to plain old C when possible) (debatable).